### PR TITLE
Add unit integ test for feature support ms teams

### PR DESCRIPTION
**Pull request by dankyalo599**
### Description
[FEATURE] Support Microsoft Teams webhooks #585
### Issues Resolved
Add unit and integTest for feature support ms teams

### Check List
- [ X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Hailong-am** [comment]
depends on https://github.com/opensearch-project/common-utils/pull/374

File: notifications/core-spi/src/test/kotlin/org/opensearch/notifications/spi/utils/ValidationHelpersTests.kt [hunk 1]
```diff
@@ -94,9 +95,16 @@ internal class ValidationHelpersTests {
...
-
+    @Test
+    fun `validator identifies webhook url as valid`() {
+        assert(isValidUrl(MICROSOFT_TEAMS_WEBHOOK_URL))
+    }
     @Test
     fun `validator identifies chime url as valid`() {
         assert(isValidUrl(CHIME_URL))
     }
+    @Test
```
Comment (zhichao-aws) [inline notifications/core-spi/src/test/kotlin/org/opensearch/notifications/spi/utils/ValidationHelpersTests.kt:106]: Functions should be separated by a blank line.

File: notifications/notifications/src/main/kotlin/org/opensearch/notifications/send/SendMessageActionHelper.kt [hunk 2]
```diff
@@ -336,6 +337,12 @@ object SendMessageActionHelper {
                 LegacyDestinationResponse.Builder().withStatusCode(400)
                     .withResponseContent("Channel type given (sns) for publishing to legacy destination not supported").build()
             }
+//            LegacyDestinationType.LEGACY_CUSTOM_WEBHOOK -> {
```
Comment (zhichao-aws) [inline notifications/notifications/src/main/kotlin/org/opensearch/notifications/send/SendMessageActionHelper.kt:340]: we can remove these comments

File: notifications/notifications/src/test/kotlin/org/opensearch/integtest/send/SendTestMessageRestHandlerIT.kt [hunk 3]
```diff
@@ -115,6 +146,38 @@ internal class SendTestMessageRestHandlerIT : PluginRestTestCase() {
         val error = sendResponse.get("error").asJsonObject
         Assert.assertNotNull(error.get("reason").asString)
     }
+    @Suppress("EmptyFunctionBlock")
```
Comment (zhichao-aws) [inline notifications/notifications/src/test/kotlin/org/opensearch/integtest/send/SendTestMessageRestHandlerIT.kt:149]: we can remove this one and only keep the first test. Besides, we shouldn't put real webhook in code

File: notifications/notifications/src/test/kotlin/org/opensearch/notifications/ObjectEqualsHelpers.kt [hunk 4]
```diff
@@ -83,6 +83,7 @@ fun verifyEquals(config: NotificationConfig, jsonObject: JsonObject) {
     when (config.configType) {
         ConfigType.SLACK -> verifyEquals((config.configData as Slack), jsonObject.get("slack").asJsonObject)
         ConfigType.CHIME -> verifyEquals((config.configData as Chime), jsonObject.get("chime").asJsonObject)
+        ConfigType.MICROSOFT_TEAMS -> verifyEquals((config.configData as Microsoft_teams), jsonObject.get("microsoft_teams").asJsonObject)
```
Comment (Hailong-am) [inline notifications/notifications/src/test/kotlin/org/opensearch/notifications/ObjectEqualsHelpers.kt:86]: same here, should be `MicrosoftTeams`

File: notifications/core/src/test/kotlin/org/opensearch/notifications/core/destinations/MicrosofTeamsDestinationTests.kt [hunk 5]
```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Comment (Hailong-am) [inline notifications/core/src/test/kotlin/org/opensearch/notifications/core/destinations/MicrosofTeamsDestinationTests.kt:5]: missing import here.

File: notifications/core/src/test/kotlin/org/opensearch/notifications/core/destinations/MicrosofTeamsDestinationTests.kt [hunk 6]
```diff
@@ -0,0 +1,91 @@
+/*
```
Comment (Hailong-am) [inline notifications/core/src/test/kotlin/org/opensearch/notifications/core/destinations/MicrosofTeamsDestinationTests.kt:1]: this is a duplicate with [MicrosoftTeamsDestinationTests.kt](https://github.com/opensearch-project/notifications/pull/638/files#diff-c2954fed5ee706eae7ade0b833d6775a2d23da2f23c167be4dc8ef0fbce0efd4), please remove it.

**Review by zhichao-aws — CHANGES_REQUESTED** [review]
Great work, but need some changes. Besides the issues I mentioned in the comments, there are still some bugs, such as not establishing a mapping in DestinationTransportProvider. The current code cannot pass the code style check (kotlint) and compilation. And it also can not work. Please fix your build and run tests locally.

BTW, you can directly add new comments to this PR, don't need to create new PR :)

**Comment by zhichao-aws** [comment]
Hi, I create a new [PR](https://github.com/opensearch-project/notifications/pull/676) to fix the build, and add more integTest. Close this one.

**Pull request closed without merging**